lib/instrumentation/pino/pino.js

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+const {
+  isApplicationLoggingEnabled,
+  isLogForwardingEnabled,
+  isMetricsEnabled,
+  isLocalDecoratingEnabled,
+  incrementLoggingLinesMetrics,
+  truncate
+} = require('../../util/application-logging')
+const semver = require('semver')
+
+module.exports = function instrument(shim) {
+  const pinoVersion = shim.require('./package.json').version
+
+  if (semver.lt(pinoVersion, '7.0.0')) {
+    shim.logger.debug('Instrumentation only supported on pino >=7.0.0.')
+    return
+  }
+
+  const tools = shim.require('./lib/tools')
+  const agent = shim.agent
+  const config = agent.config
+
+  if (!isApplicationLoggingEnabled(config)) {
+    shim.logger.debug('Application logging not enabled. Not instrumenting pino.')
+    return
+  }
+
+  const metrics = agent.metrics
+  createModuleUsageMetric(metrics)
+
+  const levelUtils = shim.require('./lib/levels')
+
+  // Create a where the keys are the numbers and the values are the labels
+  const levelMap = Object.entries(levelUtils.levels).reduce((prev, current) => {
+    const [key, value] = current
+    prev[value] = key
+    return prev
+  }, {})
```

Comment:
> This could introduce a subtle problem in case a user decided to use two labels as aliases for the same number. I think pino allows that, although I assume this is such a rare situation that perhaps not worth fretting over.

Yea I won't worry about this. I can change the variables, but was following names in MDN